### PR TITLE
Updated intro-to-clr document.

### PR DESCRIPTION
**Pull request by makcakaya**
Updated a few points in the document according to my understanding.

### CODE REVIEW

File: Documentation/intro-to-clr.md [hunk 1]
```diff
@@ -140,7 +140,7 @@ For type safety, conceptually each memory allocation is associated with a type.
 
 One the most important of these type-specific guarantees is that the visibility attributes associated with a type (and in particular with fields) are enforced.  Thus, if a field is declared to be private (accessible only by the methods of the type), then that privacy will indeed be respected by all other type-safe code.  For example, a particular type might declare a count field that represents the count of items in a table.  Assuming the fields for the count and the table are private, and assuming that the only code that updates them updates them together, there is now a strong guarantee (across all type-safe code) that the count and the number of items in the table are indeed in sync.  When reasoning about programs, programmers use the concept of type safety all the time, whether they know it or not.  The CLR elevates type-safety from being simply a programming language/compiler convention, to something that can be strictly enforced at run time.
 
-### Enforcing memory safety (Verifiable Code)
+### Enforcing Verifiable Code (Memory and Type Safety)
```
Comment (richlander) [inline Documentation/intro-to-clr.md]: "Enforcing Verifiable Code" doesn't quite make sense to me. Verifiable code is something that you arrive at by construction, not by "enforcement". My view (and I think that this also what the next paragraph by @vancem is saying) is that verifiable code is what is used to enforce type safety.

How about "Verifiable Code - Enforcing Memory and Type Safety"??

File: Documentation/intro-to-clr.md [hunk 2]
```diff
@@ -202,7 +202,7 @@ As an aside, while exceptions avoid one common error (not checking for failure),
...
+Previous to version 2.0 of the CLR, the only parameterized types were arrays.  All other containers (such as hash tables, lists, queues, etc.), all operated on a generic Object type.  The inability to create List<ElemT>, or Dictionary<KeyT, ValueT> certainly had a negative performance effect because value types needed to be boxed on entry to a collection, and explicit casting was needed on element fetch.  Nevertheless, that is not the overriding reason for adding parameterized types to the CLR.  The main reason is that **parameterized types make programming easier** by enabling static type-checking at compilation phase and enforcing type-safety.
```
Comment (richlander) [inline Documentation/intro-to-clr.md]: For clarity, which compilation phase do you mean? Source compilation (e.g. CSC) or bytecode compilation (JIT)?

The sentence doesn't quite follow for me. I think that Vance was saying that generics make programming more easy and pleasant, not that it specifically enables a set of checks. You are adding a technical conclusion that I don't think the paragraph was headed towards.  Maybe make  a paragraph for this discussion?

File: Documentation/intro-to-clr.md (hunk 1 above)
Comment (makcakaya) [inline Documentation/intro-to-clr.md]: You are right, thanks.

File: Documentation/intro-to-clr.md (hunk 2 above)
Comment (makcakaya) [inline Documentation/intro-to-clr.md]: I meant CIL to native compilation but CSC-to-CIL holds too. I will drop the change and try to add a new paragraph after I become more familiar with the project :)

**Comment by richlander** [comment]
@makcakaya Thanks for the update. Appreciated.

**Pull request merged**